### PR TITLE
Refactor to typescript

### PR DESCRIPTION
**Pull request by SharonStrats**
Converting to Typescript

### CODE REVIEW

File: src/widgets/dragAndDrop.ts [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-
-function makeDropTarget (ele, droppedURIHandler, droppedFileHandler) {
-  var dragoverListener = function (e) {
+/**
+ * @ignore
+ */
+class FileObject extends File {
+  lastModifiedDate?: Date
+}
+export function makeDropTarget (ele: HTMLElement, droppedURIHandler: any, droppedFileHandler: any) {
```
Comment (SharonStrats) [inline src/widgets/dragAndDrop.ts]: I tried to use a function type for droppedURIHandler and droppedFileHandler but I couldn't get it to work with my unit tests so I just put it back to any.  Would love to hear your suggestion on this.

File: src/widgets/dragAndDrop.ts [hunk 2]
```diff
@@ -48,30 +54,32 @@ function makeDropTarget (ele, droppedURIHandler, droppedFileHandler) {
     }
   }
 
-  var dropListener = function (e) {
+  const dropListener = function (this: any, e: DragEvent) {
```
Comment (SharonStrats) [inline src/widgets/dragAndDrop.ts]: I think I have more comments about this in the draganddrop branch, but in order to use this. you have to pass in this:any, there were a couple more ideas, but this seem to be the best option.

File: src/widgets/dragAndDrop.ts [hunk 3]
```diff
@@ -169,50 +183,50 @@ function makeDraggable (tr, obj) {
  **  successHandler(file, uploadedURI)    Called after each success upload
  **                              With file object an final URI as params
  */
-function uploadFiles (fetcher, files, fileBase, imageBase, successHandler) {
-  for (var i = 0; files[i]; i++) {
-    const f = files[i]
-    debug.log(
+export function uploadFiles (fetcher: any, files: any, fileBase: string, imageBase: string, successHandler: any) {
```
Comment (SharonStrats) [inline src/widgets/dragAndDrop.ts]: A similar thing happened here with successHandler.  files as well.  I can put this as FileList which it is, but for testing I couldn't quickly figure this out.  I tried to createElement('input') and set type to 'files' but it didn't work. Fetcher I just realized, I will look at this.

Comment (SharonStrats) [reply src/widgets/dragAndDrop.ts]: I tried to use Fetcher from rdflib, but it gave me the error .webOperation does not exist on type 'Fetcher'

**Comment by michielbdejong** [comment]
Is this ready for review?

**Comment by michielbdejong** [comment]
Travis says:
```
src/messageArea.ts(126,9): error TS2345: Argument of type 'Statement' is not assignable to parameter of type 'never'.
src/messageArea.ts(130,9): error TS2345: Argument of type 'Statement' is not assignable to parameter of type 'never'.
src/messageArea.ts(138,9): error TS2345: Argument of type 'Statement' is not assignable to parameter of type 'never'.
src/messageArea.ts(142,11): error TS2345: Argument of type 'Statement' is not assignable to parameter of type 'never'.
src/widgets/peoplePicker.ts(66,15): error TS2345: Argument of type '({ book }: { book: any; }) => void' is not assignable to parameter of type '(value: unknown) => void | PromiseLike<void>'.
  Types of parameters '__0' and 'value' are incompatible.
    Type 'unknown' is not assignable to type '{ book: any; }'.
src/widgets/peoplePicker.ts(176,29): error TS2345: Argument of type '{ toIns: Statement[]; }' is not assignable to parameter of type '{ toDel: any; toIns: any; }'.
  Property 'toDel' is missing in type '{ toIns: Statement[]; }' but required in type '{ toDel: any; toIns: any; }'.
src/widgets/peoplePicker.ts(391,13): error TS2345: Argument of type '(webIdNode: NamedNode) => any' is not assignable to parameter of type '(value: unknown) => any'.
  Types of parameters 'webIdNode' and 'value' are incompatible.
    Type 'unknown' is not assignable to type 'NamedNode'.
src/widgets/peoplePicker.ts(392,25): error TS2554: Expected 4 arguments, but got 3.
src/widgets/peoplePicker.ts(396,42): error TS2345: Argument of type '{ toIns: Statement[]; }' is not assignable to parameter of type '{ toDel: any; toIns: any; }'.
  Property 'toDel' is missing in type '{ toIns: Statement[]; }' but required in type '{ toDel: any; toIns: any; }'.
src/widgets/peoplePicker.ts(407,25): error TS2554: Expected 4 arguments, but got 3.
src/widgets/peoplePicker.ts(408,42): error TS2345: Argument of type '{ toDel: Statement[]; }' is not assignable to parameter of type '{ toDel: any; toIns: any; }'.
  Property 'toIns' is missing in type '{ toDel: Statement[]; }' but required in type '{ toDel: any; toIns: any; }'.

File: src/widgets/peoplePicker.ts [hunk 4]
```diff
@@ -379,9 +402,9 @@ export class GroupBuilder {
     })
   }
 
-  handleRemove (webIdNode) {
+  handleRemove (webIdNode: NamedNode) {
     return _event => {
-      const statement = rdf.st(this.group, ns.vcard('hasMember'), webIdNode)
```
Comment (SharonStrats) [inline src/widgets/peoplePicker.ts]: any suggestions on how to handle this?  rdf.st needs subject, predicate, object, graph but here there are only 3 arguments given, I believe it's missing the object/value.

**Comment by megoth** [comment]
@michielbdejong I don't think we should review this, it's just the work that @SharonStrats was able to do wrt migrating the remaining JS to TS. Let's revisit this after I've worked on fixing rdflib.

File: src/widgets/peoplePicker.ts (hunk 4 above)
Comment (michielbdejong) [reply src/widgets/peoplePicker.ts]: pass `undefined` (or if that doesn't work, `null`) as the fourth argument

**Comment by michielbdejong** [comment]
OK, but we stopped doing [WiP] PRs, so then I think @SharonStrats should close this PR and create an issue instead, right?

**Pull request closed without merging**